### ISSUE TITLE
private beta cluster module has non-obvious problem causing failure

### ISSUE THREAD

**Issue by ideasculptor** [terraform-google-modules/terraform-google-kubernetes-engine#304]
I have what I think is a reasonably simple config for a GKE cluster, but it always times out while waiting for deployment - I have never seen a node actually launch, or the node pool even set to a size other than 0, despite the fact that I have explicitly configured it to an initial size of 1 and told it not to delete the default node pool.  The error message provides no insight as to the failure at all, and neither do the logs.

```
module.gke.google_container_cluster.primary: Still creating... [50m51s elapsed]
module.gke.google_container_cluster.primary: Still creating... [51m1s elapsed]
module.gke.google_container_cluster.primary: Still creating... [51m11s elapsed]
module.gke.google_container_cluster.primary: Still creating... [51m21s elapsed]
module.gke.google_container_cluster.primary: Still creating... [51m31s elapsed]

Error: Error waiting for creating GKE cluster: Failed to create cluster

  on .terraform/modules/gke/terraform-google-modules-terraform-google-kubernetes-engine-ab02f24/modules/beta-private-cluster/cluster.tf line 22, in resource "google_container_cluster" "primary":
  22: resource "google_container_cluster" "primary" {
```

```
locals {
  project_id         = data.terraform_remote_state.service_project.outputs.project_id
  network_project_id = data.terraform_remote_state.vpc.outputs.project_id
  network            = data.terraform_remote_state.vpc.outputs.network_name
  subnetwork         = "${var.environment}-${var.region}-${var.subnet_short_name}"
  ip_range_pods      = "${local.subnetwork}-${var.pods_range_short_name}"
  ip_range_services  = "${local.subnetwork}-${var.services_range_short_name}"
}

module "gke" {
  source = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster"

  project_id                        = local.project_id
  network_project_id                = local.network_project_id
  registry_project_id               = local.project_id
  name                              = "gke-${var.environment}"
  description                       = var.description
  region                            = "us-central1"
  regional                          = false
  zones                             = ["us-central1-a"]
  network                           = local.network
  subnetwork                        = local.subnetwork
  ip_range_pods                     = local.ip_range_pods
  ip_range_services                 = local.ip_range_services
  http_load_balancing               = true
  horizontal_pod_autoscaling        = true
  kubernetes_dashboard              = true
  create_service_account            = true
  grant_registry_access             = true
  initial_node_count                = 1
  issue_client_certificate          = false
  kubernetes_version                = "latest"
  master_authorized_networks_config = [{
    cidr_blocks = [{
      cidr_block = "98.232.147.100/32"
      display_name = "home"
    }]
  }]
  remove_default_node_pool          = false

  default_max_pods_per_node         = 110
  deploy_using_private_endpoint     = true

# necessary to list public addr in authorized_networks
  enable_private_endpoint           = false 
  enable_private_nodes              = true
  enable_vertical_pod_autoscaling   = false
  istio                             = true
  master_ipv4_cidr_block            = "10.0.0.0/28
}
```

There's nothing of any utility in the logs, but if you want to take a look, give me a service account address and I'll just give you access.  It's all throwaway development infrastructure anyway, so risk free.

**Comment by ideasculptor** [comment]
I have no reason to think it is a permissions problem - I had to fight my way through a bunch of those already, and eventually tore my entire infrastructure down and recreated it from scratch just to make sure there were no problems left behind by the old project attachment module.  The fact that it no longer complains about a lack of permissions would imply that I finally got that all correct.  I also have both the required apis enabled.

**Comment by ideasculptor** [comment]
I'll eventually try creating a non-default node-pool and see if it comes up all the way when I do that, but since each test takes the better part of an hour to complete, it's not going to happen until tomorrow.

**Comment by ideasculptor** [comment]
non-default node pool didn't help.  I'm assuming the fact that I am not running terraform within the VPC is the source of the problem? I was hoping it would work from my local machine, so long as it was added to the master_authorized_networks_config, but I guess not, so I'll have to set up a machine to run it from and see if that fixes it.

**Comment by morgante** [comment]
This might be related to overall GKE [issues](https://status.cloud.google.com/).

**Comment by ideasculptor** [comment]
Oh, that’d be nice. I love it when it turns out not to be my fault ;-) Thanks for pointing it out, so I don’t spend more time on it until I see the alerts clear.

**Comment by ideasculptor** [comment]
Is there some way to actually reach support on this issue?  Google claims to have resolved the issue, except for a 'small number of projects' and we should reach out to support if still having trouble.  I've had a 'cluster' that finally showed up sometime on Saturday, but with an error alert that says there is a network peering operation in progress.  I cannot destroy the cluster because it is blocked on that peering operation.  I cannot launch a replacement cluster for the same reason.  I clicked the 'send feedback' link on the GKE dashboard and sent a request for support over 12 hours ago, but I have yet to even receive confirmation that my request has been received.  

I first attempted to create the cluster on friday evening.  50+ hours later, I am still blocked on a broken system. If the company is going to declare a problem to be fixed, shouldn't it actually have to be fixed?  Did the company just declare it fixed when it ran out of available hours in the SLA, despite not actually fixing the problem?  I'm supposed to have faith that this kind of thing isn't going to happen to production resources?
